flash/tabular/classification/data/data.py

```diff
@@ -58,210 +54,142 @@ def __init__(
         self.std = std
         self.codes = codes
         self.target_codes = target_codes
-        self.num_classes = num_classes
         self.is_regression = is_regression
 
-    def get_state_dict(self, strict: bool = False) -> Dict[str, Any]:
-        return {
-            "cat_cols": self.cat_cols,
-            "num_cols": self.num_cols,
-            "target_col": self.target_col,
-            "mean": self.mean,
-            "std": self.std,
-            "codes": self.codes,
-            "target_codes": self.target_codes,
-            "classes": self.num_classes,
-            "num_classes": self.num_classes,
-            "is_regression": self.is_regression,
-        }
-
-    @classmethod
-    def load_state_dict(cls, state_dict: Dict[str, Any], strict: bool = True) -> 'Preprocess':
-        return cls(**state_dict)
+        self.set_state(LabelsState(classes))
+        self.num_classes = len(classes)
 
-    def common_load_data(self, df: DataFrame, dataset: AutoDataset):
+    def common_load_data(
+        self,
+        df: DataFrame,
+        dataset: Optional[Any] = None,
+    ):
         # impute_data
         # compute train dataset stats
         dfs = _pre_transform([df], self.num_cols, self.cat_cols, self.codes, self.mean, self.std, self.target_col,
                              self.target_codes)
 
         df = dfs[0]
 
-        dataset.num_samples = len(df)
+        if dataset is not None:
```

Comment:
```suggestion
        if dataset is not None and hasattr(dataset, 'num_samples'):
```

Reply:
We are setting the attribute here, so it doesn't need to have it beforehand